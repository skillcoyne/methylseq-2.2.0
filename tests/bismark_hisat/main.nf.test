nextflow_pipeline {

    name "bismark_hisat"
    script "main.nf"

    test("Single End") {
        when {
            params {
                aligner = "bismark_hisat"
                save_reference = true
                outdir = "$outputDir"
            }
        }

        then {
            assert workflow.success
            assert workflow.trace.tasks().size() == 26
            // FIXME assert snapshot(workflow).match()
        }
    }

    test("Single End with index") {
        when {
            params {
                aligner = "bismark_hisat"
                // Generated by previous test
                bismark_index = "$outputDir/bismark_hisat/reference_genome/BismarkIndex/"
                local_alignment = true
                outdir = "$outputDir"
            }
        }

        then {
            assert workflow.failed
            assert workflow.trace.tasks().size() == 6
            // FIXME assert snapshot(workflow).match()
        }
    }

    test("Single End RRBS") {
        when {
            params {
                aligner = "bismark_hisat"
                save_reference = true
                skip_trimming = true
                rrbs = true
                outdir = "$outputDir"
            }
        }

        then {
            assert workflow.success
            assert workflow.trace.tasks().size() == 22
            // FIXME assert snapshot(workflow).match()
        }
    }

    test("Single End RRBS with index") {
        when {
            params {
                aligner = "bismark_hisat"
                skip_trimming = true
                rrbs = true
                // Generated by previous test
                bismark_index = "$outputDir/bismark_hisat/reference_genome/BismarkIndex/"
                local_alignment = true
                outdir = "$outputDir"
            }
        }

        then {
            assert workflow.failed
            // FIXME Flakes between 3 and 4
            // assert workflow.trace.tasks().size() == 4
            // FIXME assert snapshot(workflow).match()
        }
    }

}
