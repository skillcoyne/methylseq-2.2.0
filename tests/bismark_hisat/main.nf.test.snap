{
    "Single End RRBS with index": {
        "content": [
            {
                "stderr": [

                ],
                "errorReport": "",
                "exitStatus": 0,
                "out": {
                    
                },
                "failed": false,
                "stdout": [
                    "",
                    "WARN: Found unexpected parameters:",
                    "* --outputDir: /home/emiller/src/nf-core/methylseq/.nf-test/tests/a66fb05f140c2a514788a6c37ecc9fe6/output",
                    "* --baseDir: /home/emiller/src/nf-core/methylseq",
                    ""
                ],
                "errorMessage": "",
                "trace": {
                    "tasksFailed": 3,
                    "tasksCount": 28,
                    "tasksSucceeded": 25
                },
                "name": "workflow",
                "success": true
            }
        ],
        "timestamp": "2022-11-04T03:25:15+0000"
    },
    "Single End with index": {
        "content": [
            {
                "stderr": [

                ],
                "errorReport": "",
                "exitStatus": 0,
                "out": {
                    
                },
                "failed": false,
                "stdout": [
                    "",
                    "WARN: Found unexpected parameters:",
                    "* --outputDir: /home/emiller/src/nf-core/methylseq/.nf-test/tests/2f58fc414e3af41c7dbb50ffdb77f813/output",
                    "* --baseDir: /home/emiller/src/nf-core/methylseq",
                    ""
                ],
                "errorMessage": "",
                "trace": {
                    "tasksFailed": 3,
                    "tasksCount": 34,
                    "tasksSucceeded": 31
                },
                "name": "workflow",
                "success": true
            }
        ],
        "timestamp": "2022-11-04T03:25:15+0000"
    },
    "Single End": {
        "content": [
            {
                "stderr": [

                ],
                "errorReport": "",
                "exitStatus": 0,
                "out": {
                    
                },
                "failed": false,
                "stdout": [
                    "",
                    "WARN: Found unexpected parameters:",
                    "* --outputDir: /home/emiller/src/nf-core/methylseq/.nf-test/tests/f06ee2f2f3cd9b2459646a3f5665c50d/output",
                    "* --baseDir: /home/emiller/src/nf-core/methylseq",
                    ""
                ],
                "errorMessage": "",
                "trace": {
                    "tasksFailed": 3,
                    "tasksCount": 35,
                    "tasksSucceeded": 32
                },
                "name": "workflow",
                "success": true
            }
        ],
        "timestamp": "2022-11-04T03:25:15+0000"
    },
    "Single End RRBS": {
        "content": [
            {
                "stderr": [

                ],
                "errorReport": "",
                "exitStatus": 0,
                "out": {
                    
                },
                "failed": false,
                "stdout": [
                    "",
                    "WARN: Found unexpected parameters:",
                    "* --outputDir: /home/emiller/src/nf-core/methylseq/.nf-test/tests/129e549c12043de530474cb9e08f8b7/output",
                    "* --baseDir: /home/emiller/src/nf-core/methylseq",
                    ""
                ],
                "errorMessage": "",
                "trace": {
                    "tasksFailed": 3,
                    "tasksCount": 29,
                    "tasksSucceeded": 26
                },
                "name": "workflow",
                "success": true
            }
        ],
        "timestamp": "2022-11-04T03:25:15+0000"
    }
}
